File: kit_padrao_pdf/empresa.py
# -*- coding: utf-8 -*-
"""
Identidade da empresa para os geradores ReportLab.

Espelho de `core/domain/empresa.ts`. Os dois arquivos precisam andar juntos:
o Python nao consegue importar TypeScript, entao a marca existe duas vezes e
divergir os dois faz o deck sair com um nome e o contrato com outro.

Marca e identidade juridica sao blocos separados pelo mesmo motivo que no lado
TypeScript. Enquanto `PENDENTE` for True, razao social e CNPJ continuam sendo o
par da empresa anterior, de proposito — trocar so o nome deixaria o CNPJ de
outra pessoa juridica embaixo da marca nova.

Sem acento: os geradores usam fontes que nem sempre trazem a tabela completa, e
o projeto ja escreve todo texto de PDF em ASCII por isso.
"""

MARCA = "Global Company"
MARCA_ASSINATURA = "Global Company — Inteligencia Tecnica para Gestao Educacional"

RAZAO_SOCIAL = "GLOBAL SERVICES COMPANY LTDA"
CNPJ = "26.137.996/0001-75"
ENDERECO = "Pe. Orthon Vieira Lima, S/N, Centro"
CIDADE = "Santa Maria da Vitoria"
UF = "BA"
CEP = "47640-058"
TELEFONE = "(77) 9700-5880"
EMAIL = "globalconsultorias@icloud.com"

# Vazio enquanto nao houver dominio proprio. `linhas_de_contato` omite o que
# estiver vazio — melhor faltar a linha que imprimir contato que nao atende.
SITE = ""

# Falta CPF e RG do socio-administrador para a qualificacao das partes.
PENDENTE = True


def linhas_de_contato():
    """As linhas de contato que existem, na ordem de leitura."""
    linhas = []
    if TELEFONE:
        linhas.append("Tel: " + TELEFONE)
    if EMAIL:
        linhas.append("E-mail: " + EMAIL)
    if SITE:
        linhas.append("Site: " + SITE)
    return linhas

File: core/domain/empresa.ts
/**
 * Identidade da empresa, num lugar só.
 *
 * Antes destes valores existirem aqui, a marca aparecia escrita à mão em onze
 * lugares — dois serviços de contrato, cinco geradores de PDF, a tela de
 * ajustes — e a troca de "Rocha Prime" por "Global Company" foi exatamente o
 * tipo de mudança que esse espalhamento torna cara e arriscada.
 *
 * ## Marca e identidade jurídica são coisas separadas
 *
 * `MARCA` é como a empresa se apresenta: capa de deck, rodapé de relatório,
 * cabeçalho da interface. Muda quando o marketing quer.
 *
 * `EMPRESA` é quem assina contrato. Vai na minuta de assessoria, no parecer de
 * inexigibilidade e na proposta comercial — peças que entram em processo
 * administrativo municipal. Errar um dígito de CNPJ aqui não é bug de tela.
 *
 * Por isso os dois blocos não se misturam, e por isso o bloco jurídico carrega
 * `pendente`: enquanto o dado novo não chega, o gerador tem como saber que o
 * que ele tem em mãos não descreve mais a empresa e agir de acordo — omitir a
 * linha em vez de imprimir contato morto.
 *
 * O espelho em Python vive em `kit_padrao_pdf/empresa.py`. Os geradores
 * ReportLab não conseguem importar TypeScript, então os dois arquivos precisam
 * andar juntos — mesma disciplina da lista `COMPLEMENTOS` do empacotador.
 */

/** Como a empresa se apresenta. Forma curta, a que vai na arte. */
export const MARCA = "Global Company";

/** Uma linha de posicionamento, usada em capa de deck e assinatura de e-mail. */
export const MARCA_ASSINATURA =
  "Global Company — Inteligência Técnica para Gestão Educacional";

/** Quem assina pela empresa. Vai na qualificação das partes e na procuração. */
export interface RepresentanteLegal {
  nome: string;
  cpf: string;
  rg: string;
  orgaoExpedidor: string;
  nacionalidade: string;
  estadoCivil: string;
  qualificacao: string;
}

export interface IdentidadeJuridica {
  razaoSocial: string;
  cnpj: string;
  endereco: string;
  cidade: string;
  uf: string;
  cep: string;
  telefone: string;
  /** Vazio enquanto o domínio novo não existe. Gerador omite linha vazia. */
  email: string;
  /** Vazio enquanto o domínio novo não existe. Gerador omite linha vazia. */
  site: string;
  representante: RepresentanteLegal;
  /**
   * `true` significa: falta dado para este bloco assinar contrato sozinho.
   *
   * Hoje o que falta é o CPF e o RG do sócio-administrador — a Receita publica
   * o nome no quadro societário, não os documentos. Sem eles a qualificação
   * das partes na minuta sai incompleta.
   *
   * Para encerrar a pendência: preencher `representante.cpf`, `.rg` e
   * `.orgaoExpedidor`, e virar para `false`.
   */
  pendente: boolean;
}

/**
 * Confirmado na Receita Federal em 2026-08-05 (CNPJ 26.137.996/0001-75,
 * situação cadastral ATIVA desde 2016-09-09).
 */
export const EMPRESA: IdentidadeJuridica = {
  razaoSocial: "GLOBAL SERVICES COMPANY LTDA",
  cnpj: "26.137.996/0001-75",
  endereco: "Pe. Orthon Vieira Lima, S/N, Centro",
  cidade: "Santa Maria da Vitória",
  uf: "BA",
  cep: "47640-058",
  telefone: "(61) 98155-1533",
  email: "globalconsultorias@icloud.com",
  site: "",
  representante: {
    nome: "ADRIEL PEREIRA TAVARES",
    cpf: "",
    rg: "",
    orgaoExpedidor: "",
    nacionalidade: "brasileiro",
    estadoCivil: "solteiro",
    qualificacao: "Sócio-Administrador",
  },
  pendente: true,
};

/** Endereço em uma linha, como sai impresso na qualificação das partes. */
export function enderecoCompleto(empresa: IdentidadeJuridica = EMPRESA): string {
  return empresa.cep
    ? `${empresa.endereco}, CEP: ${empresa.cep}`
    : empresa.endereco;
}

/** As linhas de contato que existem — sem buracos e sem contato morto. */
export function linhasDeContato(
  empresa: IdentidadeJuridica = EMPRESA,
): string[] {
  return [
    empresa.telefone && `Tel: ${empresa.telefone}`,
    empresa.email && `E-mail: ${empresa.email}`,
    empresa.site && `Site: ${empresa.site}`,
  ].filter((linha): linha is string => Boolean(linha));
}
